testing/e2e/clients/aks.go

```diff
@@ -32,13 +38,22 @@ func PrivateClusterOpt(mc *armcontainerservice.ManagedCluster) error {
 	return nil
 }
 
+func LoadAks(id arm.ResourceID) *aks {
+	return &aks{
+		id:             id.String(),
+		name:           id.Name,
```

Comment:
I prefer to keep things on our client structs scoped to what we need.

`arm.Resource` contains a number of extra fields that are sometimes blank and won't be used at all. Scoping down to only what we need and will be populated makes types easier to work with because you don't have to know what's available and what's not, everything that's there is populated and useful.

Reply:
^That's for why not arm.ResourceID. For why not just have a field with id it's a lot more work to parse the id every time to get things like resource name or subscription than just storing them.

This also matches what the ARM structs do.

<img width="427" alt="image" src="https://github.com/Azure/aks-app-routing-operator/assets/33605867/8c543a7a-fb16-4b66-b593-0923e474c78d">

Reply:
makes sense!